### PR TITLE
Observable as Hook

### PR DESCRIPTION
**Pull request by nikonikoniko**
Integrate observable as hook into the react world.

This way each block component can offer an observable as to when it is available.

### CODE REVIEW

File: src/frontend/containers/block/addBlock.tsx [hunk 1]
```diff
@@ -1,10 +1,13 @@
-import React from "react";
-import * as dbTypes from "@db/types";
-import { promiseAsHook } from "@frontend/utils";
-
 import * as note from "./note";
+import * as dbTypes from "@db/types";
+import {
+  // promiseAsHook,
+  observableAsHook,
```
Comment (tlmn) [inline src/frontend/containers/block/addBlock.tsx:5]: just a neurotic naming question: wouldn't the canoncial name rather be `useObservable`?

File: src/frontend/utils/index.ts [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
+import { Observable } from "rxjs";
 
 // takes a promise, and returns a react hook for it.
+// use like this:
+//
+// export const isAvailable = (db: dbTypes.LoadedDb): Promise<boolean> => Promise.resolve(true);
+//
+// const maybeAddNote = promiseAsHook(note.isAvailable(db)) && "it's available!";
+//
 export const promiseAsHook = <T>(p: Promise<T>) => {
```
Comment (tlmn) [inline src/frontend/utils/index.ts:11]: same neuroticism: `usePromise()`?

**Comment by tlmn** [comment]
looks good to me :) general question though: why not use `rxdb-hooks`?

File: src/frontend/utils/index.ts (hunk 2 above)
Comment (nikonikoniko) [reply src/frontend/utils/index.ts:11]: I like it, `usePromise` and `useObservable`

**Comment by nikonikoniko** [comment]
> looks good to me :) general question though: why not use `rxdb-hooks`?

Just in case the `block` wants to use more complicated functionality.  these are created primarily for the `Block.isAvailable` function.  The block might only be available if an external service is available, _and_ you have the data to operate on already in the database.  

In this case, the Block's responsibility is to expose it's availability either as a promise or observable, which may have implementations outside of `rxdb-hooks`.

**Pull request merged**